### ISSUE TITLE
Latex blocks should work even when not preceded by empty line

### ISSUE THREAD

**Issue by tomtung** [tamlok/vnote#754]
OS Version (操作系统): Windows

VNote Version (Help - About VNote - Version: )(VNote版本): 2.4

Symptoms (表现):
Latex blocks only work when they are preceded by an empty line. This behavior is not consistent with other blocks (e.g. code, table, etc.).

How to Repro (如何复现):
For the following Markdown document:
`````
test
```
test1
```

test
| test2 |
| ----- |

test
$$
test3
$$

$$
test4
$$
`````
"test1" is rendered as a code block, "test2" is rendered as a table, but "test3" is not rendered as a formula block. It works only when preceded by an empty line (e.g. "test4").

**Comment by tamlok** [comment]
VNote use [markdown-it-texmath](https://github.com/goessner/markdown-it-texmath) plugin for this. We need it to be fixed in upstream first.

**Comment by tomtung** [comment]
I think your fork of peg-markdown-highlight (https://github.com/tamlok/peg-markdown-highlight) isn't working properly either. For example, under `test/`, changing the content of `test.md` to:
```
test

$$
test3
$$
```
and run `run_test.sh`, it will print out
```
[6-17] 27 0x7ffff4df7810: DISPLAYFORMULA
```
but if you change the content of `test.md` to:
```
test
$$
test3
$$
```
and run `run_test.sh`, it won't print anything.

**Comment by tamlok** [comment]
Yes, your are right. It is easy to fix peg-markdown-highlight. But edit mode should be aligned to read mode. I am not familiar with JS and markdown-it plugin development. Hence once someone fix that in texmath plugin, I will get it fixed in peg-markdown-highlight.

**Comment by tomtung** [comment]
Reported upstream https://github.com/goessner/markdown-it-texmath/issues/16
